language/move-prover/bytecode/src/global_invariant_instrumentation_v2.rs

```diff
@@ -476,8 +476,15 @@ impl<'a> Instrumenter<'a> {
         inv_set: &BTreeSet<GlobalId>,
         prop_kind: PropKind,
     ) {
+        let global_env = self.builder.fun_env.module_env.env;
         for (loc, mid, cond) in xlated_invariants {
             if inv_set.contains(mid) {
+                // This can be hard to debug
+                if prop_kind == PropKind::Assume {
+                    if global_env.get_global_invariant(*mid).unwrap().kind == ConditionKind::InvariantUpdate {
+                        panic!("Not allowed to assume update invariant");
```

Comment:
I think it might be good to report this through `GlobalEnv` instead of just panic here, so sth like:

```rust
global_env.error(
  loc,
  "Cannot assume an update invariant",
);
```

Reply:
This is purely an internal error, indicating a bug in the invariant processing.  I don't want to report it to the user as something they think they can deal with.

What is our convention for reporting system errors?  Is it something other than "panic!"?

Reply:
Ah, I see. I misunderstood the code. Since this is an internal invariant violation, we can keep the `panic!` here. That is what we have been using in the code.